### PR TITLE
Add ids to Styleguide components and HTML elements

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### What problem is this solving?

In order to do the e2e tests, the elements need to have an identifier. This PR does it.

#### How should this be manually tested?

[Workspace](https://testid--vtexgame1.myvtex.com/cart)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] [Linked this PR to a Clubhouse story (if applicable).](https://app.clubhouse.io/vtex/story/24034/adicionar-atributo-testid-em-elementos-da-ui)
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
✔️ | Technical improvements

### CODE REVIEW

File: react/ProductVariations.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const ProductVariations: FunctionComponent = () => {
     <div className={`c-muted-1 f6 lh-copy ${opaque(item.availability)}`}>
       {item.skuSpecifications.map((spec: SKUSpecification, idx: number) => {
         return (
-          <div key={idx}>
+          <div id={`${idx}`} key={idx}>
```
Comment (marcoskwkm) [inline react/ProductVariations.tsx]: Not a good idea to use an index as `id`. Maybe use `${spec.fieldName}-specification` or something similar.
Also please change the `key` prop as well.

**Review by marcoskwkm — CHANGES_REQUESTED** [review]

File: react/ProductVariations.tsx [hunk 2]
```diff
@@ -10,7 +10,10 @@ const ProductVariations: FunctionComponent = () => {
     <div className={`c-muted-1 f6 lh-copy ${opaque(item.availability)}`}>
       {item.skuSpecifications.map((spec: SKUSpecification, idx: number) => {
         return (
-          <div id={`${idx}`} key={idx}>
+          <div
+            id={`${idx}-${spec.fieldName}-specification`}
```
Comment (marcoskwkm) [inline react/ProductVariations.tsx]: `idx` is not an identifier, it's the index in the `skuSpecifications` array. For that reason the `id` you picked can still cause collisions. It's better to use `item.id` and concatenate it with the `spec.fieldName`.

**Review by marcoskwkm — CHANGES_REQUESTED** [review]

File: react/ProductVariations.tsx (hunk 2 above)
Comment (jeffersontuc) [reply react/ProductVariations.tsx]: Gotcha!

File: react/ProductVariations.tsx [hunk 3]
```diff
@@ -8,10 +8,10 @@ const ProductVariations: FunctionComponent = () => {
 
   return item.skuSpecifications && item.skuSpecifications.length > 0 ? (
     <div className={`c-muted-1 f6 lh-copy ${opaque(item.availability)}`}>
-      {item.skuSpecifications.map((spec: SKUSpecification, idx: number) => {
+      {item.skuSpecifications.map((spec: SKUSpecification) => {
         return (
           <div
-            id={`${idx}-${spec.fieldName}-specification`}
+            id={`${item.id}-${spec.fieldName}-specification`}
             key={`${spec.fieldName}-key`}
```
Comment (marcoskwkm) [inline react/ProductVariations.tsx]: Don't need `-key` here.

**Review by marcoskwkm — APPROVED** [review]

**Review by marcoskwkm — APPROVED** [review]

**Pull request merged**